extension/src/extension.ts

```diff
@@ -44,24 +46,38 @@ let running: boolean = false;
 // your extension is activated the very first time the command is executed
 export function activate(context: ExtensionContext) {
     activateTelemetry(context);
+    testResourceManager.refresh();
     const codeLensProvider = new JUnitCodeLensProvider(onDidChange, testResourceManager, logger);
     context.subscriptions.push(languages.registerCodeLensProvider(Configs.LANGUAGE, codeLensProvider));
     const testReportProvider: TestReportProvider = new TestReportProvider(testResourceManager);
     context.subscriptions.push(workspace.registerTextDocumentContentProvider(TestReportProvider.scheme, testReportProvider));
+    const testExplorer = new TestExplorer(context, testResourceManager);
+    context.subscriptions.push(window.registerTreeDataProvider(Constants.TEST_EXPLORER_VIEW_ID, testExplorer));
+    testResourceManager.onDidChangeTestStorage((e) => {
+        testExplorer.refresh();
+    });
 
     workspace.onDidChangeTextDocument((event) => {
         const uri = event.document.uri;
         testResourceManager.setDirty(uri);
         // onDidChange.fire();
     });
 
+    workspace.onDidSaveTextDocument((document) => {
+        const uri = document.uri;
+        testResourceManager.setDirty(uri);
+        onDidChange.fire();
```

Comment:
Who is listening to this event?

Reply:
junitCodelensProvider

Reply:
found that sometimes when editing the test code, codelens wouldn't update. So i add this to ensure codelens are refreshed by sure. see #36